cpp/daal/include/services/internal/sycl/buffer_utils_sycl.h

```diff
@@ -83,23 +128,71 @@ class BufferCopier
             DAAL_ASSERT_UNIVERSAL_BUFFER_TYPE(srcUnivers, T);
             DAAL_ASSERT_UNIVERSAL_BUFFER_TYPE(dstUnivers, T);
 
-            auto src = srcUnivers.get<T>().toSycl(status);
-            DAAL_CHECK_STATUS_RETURN_VOID_IF_FAIL(status);
+            const auto & src_buffer = srcUnivers.get<T>();
+            const auto & dst_buffer = dstUnivers.get<T>();
+
+            DAAL_ASSERT(src_buffer.size() >= srcOffset + count);
+            DAAL_ASSERT(dst_buffer.size() >= dstOffset + count);
+
+#ifdef DAAL_SYCL_INTERFACE_USM
```

Comment:
Split USM/non-USM cases into different function or functors, as it's done for `AllocateUSMBacked` and `AllocateVanilla`.